opcua/server/history_sql.py

```diff
@@ -231,6 +234,25 @@ def read_event_history(self, source_id, start, end, nb_values, evfilter):
     def _get_table_name(self, node_id):
         return str(node_id.NamespaceIndex) + '_' + str(node_id.Identifier)
 
+    def _get_event_fields(self, evtypes):
+        """
+        Get all fields from the event types that are to be historized
+        Args:
+            evtypes: List of event type nodes
+
+        Returns: List of fields for all event types
+
+        """
+        # get all fields from the event types that are to be historized
+        ev_aggregate_fields = []
+        for event_type in evtypes:
+            ev_aggregate_fields.extend((events.get_event_properties_from_type_node(event_type)))
+
+        ev_fields = []
+        for field in set(ev_aggregate_fields):
+            ev_fields.append(field.get_display_name().Text.decode(encoding='utf-8'))
+        return ev_fields
```

Comment:
Use list comprehension here to simplify and shorten code.

And are you sure you do not get duplicates?